### PR TITLE
TEST-41747 - Motr DI

### PR DESCRIPTION
**Pull request by pranav-sahasrabudhe**
# Problem Statement
- TEST-41747 Parity corruption and detection with M0cp and M0cat and Error_Detection.py script 

# Design
-  Parity corruption and detection with M0cp and M0cat and Error_Detection.py script 

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: comptests/motr/test_motr_di.py [hunk 1]
```diff
@@ -107,36 +132,128 @@ def m0cp_corrupt_data_m0cat(self, layout_ids, bsize_list, count_list, offsets):
...
+                        offset=offset,
+                    )
+                    self.motr_obj.cat_cmd(
+                        b_size, cnt_c, object_id, layout, outfile, node, client_num
+                    )
                     self.motr_obj.md5sum_cmd(infile, outfile, node, flag=True)
                     self.motr_obj.unlink_cmd(object_id, layout, node, client_num)
 
             logger.info("Stop: Verify multiple m0cp/cat operation")
+        return True  # Todo: return status to be worked as per responses
```
Comment (pragamj) [inline comptests/motr/test_motr_di.py:217]: We won't require return statement as in the individual function of m0cp/m0cat/md5sum,unlink have assert.

File: comptests/motr/test_motr_di.py [hunk 2]
```diff
@@ -107,36 +132,128 @@ def m0cp_corrupt_data_m0cat(self, layout_ids, bsize_list, count_list, offsets):
...
+
+        for node_pod in node_pod_dict:
+            for b_size, (cnt_c, cnt_u), layout, offset in zip(
+                bsize_list, count_list, layout_ids, offsets
+            ):
+                # Create file for m0cp cmd
+                self.motr_obj.dd_cmd(b_size, cnt_c, infile, node_pod)
+                # Create object
+                self.motr_obj.cp_cmd(
+                    b_size, cnt_c, object_id, layout, infile, node_pod, 0
```
Comment (pragamj) [inline comptests/motr/test_motr_di.py:239]: client count is set to 0 ?

File: comptests/motr/test_motr_di.py [hunk 3]
```diff
@@ -107,36 +132,128 @@ def m0cp_corrupt_data_m0cat(self, layout_ids, bsize_list, count_list, offsets):
...
+        node_pod_dict = self.motr_obj.get_node_pod_dict()
+        object_id = (
+            str(self.system_random.randint(1, 1024 * 1024))
+            + ":"
+            + str(self.system_random.randint(1, 1024 * 1024))
+        )
+
+        for node_pod in node_pod_dict:
+            for b_size, (cnt_c, cnt_u), layout, offset in zip(
+                bsize_list, count_list, layout_ids, offsets
```
Comment (pragamj) [inline comptests/motr/test_motr_di.py:233]: we can remove offsets , if its not used

**Review by pragamj — CHANGES_REQUESTED** [review]
Added few review comments.

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**